adal/src/main/java/com/microsoft/aad/adal/ADALError.java

```diff
@@ -120,6 +120,11 @@
     SERVER_ERROR("Server returned an error"),
 
     /**
+     * Server returned the 500/503/504 error.
+     */
+    SERVER_NOT_RESPONDING("Server returned the 500/503/504 error"),
```

Comment:
If we get the status code back then it means that the server has responded, but with error. SERVER_NOT_RESPONDING would apply to connection timeout or refused.